tough/src/schema/key.rs

```diff
@@ -12,71 +14,113 @@ use std::collections::HashMap;
 use std::fmt;
 use std::str::FromStr;
 
+/// Serializes signing keys as defined by the TUF specification. All keys have the format
+/// ```json
+///  { "keytype" : "KEYTYPE",
+///     "scheme" : "SCHEME",
+///     "keyval" : "KEYVAL"
+///  }
+/// ```
+/// where:
+/// ```text
+///    KEYTYPE is a string denoting a public key signature system, such
+///    as RSA or ECDSA.
+///
+///    SCHEME is a string denoting a corresponding signature scheme.  For
+///    example: "rsassa-pss-sha256" and "ecdsa-sha2-nistp256".
+///
+///    KEYVAL is a dictionary containing the public portion of the key.
```

Comment:
Can we be a bit more specific about what has to be in the dictionary, or link somewhere, or point to a type below that represents its contents?

Reply:
The spec is not as crisp as would be handy around here, but took a stab at describing the KEYVAL 'map', which in all cases just requires a `public` field.